From 4bbfaee8265315fde8b4c2d524d50823c449e18f Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Mon, 7 Nov 2016 22:52:45 +0100
Subject: [PATCH] gdk: Ensure surface buffer scale is set

GDK will only set the buffer scale for the surfaces it creates when
attaching an image or when using an egl window window. Neither will
happen for native clutter applications, thus directly set the buffer
scale from clutter in that case (so it's in sync with the rendering
scale)

Signed-off-by: Sjoerd Simons <sjoerd@luon.net>

https://bugzilla.gnome.org/show_bug.cgi?id=769190
---
 clutter/gdk/clutter-stage-gdk.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/clutter/gdk/clutter-stage-gdk.c b/clutter/gdk/clutter-stage-gdk.c
index 742de58..b4cc4d4 100644
--- a/clutter/gdk/clutter-stage-gdk.c
+++ b/clutter/gdk/clutter-stage-gdk.c
@@ -227,7 +227,14 @@ clutter_stage_gdk_wayland_surface (ClutterStageGdk *stage_gdk)
 
   if (!stage_gdk->foreign_window ||
       gdk_window_get_window_type (stage_gdk->window) != GDK_WINDOW_CHILD)
-    return gdk_wayland_window_get_wl_surface (stage_gdk->window);
+    {
+      struct wl_surface *surface;
+
+      surface = gdk_wayland_window_get_wl_surface (stage_gdk->window);
+      wl_surface_set_buffer_scale (surface,
+                                   gdk_window_get_scale_factor (stage_gdk->window));
+      return surface;
+    }
 
   if (stage_gdk->clutter_surface)
     return stage_gdk->clutter_surface;
-- 
2.10.2

